cfgov/v1/management/commands/archive_pages.py

```diff
@@ -0,0 +1,124 @@
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            "url_path",
+            nargs="+",
+            type=path_without_leading_trailing_slashes,
+            help=(
+                "The URL path (without leading /) of a filterable list page "
+                "whose content should be archived if it was published the "
+                "specified number of years, months, and days ago. "
+                "The filerable list page at the URL will not be archived."
+            ),
+        )
+        parser.add_argument(
+            "--years",
+            type=int,
+            default=2,
+            help=(
+                "Archive content published this number of years ago. "
+                "Combines with --months --days. Default: 2"
+            ),
+        )
+        parser.add_argument(
+            "--months",
+            type=int,
+            default=0,
+            help=(
+                "Archive content published this number of months ago. "
+                "Combines with --years --days. Default: 0"
+            ),
+        )
+        parser.add_argument(
+            "--days",
+            type=int,
+            default=0,
+            help=(
+                "Archive content published this number of days ago. "
+                "Combines with --years --months. Default: 0"
+            ),
+        )
+        parser.add_argument(
+            "--by-published-date",
+            choices=["first", "last"],
+            default="first",
+            help=(
+                "Archive based on either first or last published date. "
+                "Default: first"
+            ),
+        )
+
+    def handle(self, *args, **options):
+        url_paths = options["url_path"]
+
+        # Get the current date/time and then get our cutoff date for archiving
+        # based on it.
+        archived_at = timezone.now()
+        cutoff_date = archived_at - relativedelta(
+            years=options["years"],
+            months=options["months"],
+            days=options["days"],
+        )
+
+        # Construct a Q object to filter on based on this command-line argument
+        if options["by_published_date"] == "last":
+            published_date_filter = Q(last_published_at__lt=cutoff_date)
+        else:
+            published_date_filter = Q(first_published_at__lt=cutoff_date)
+
+        # We"ll use Wagtail"s page routing to resolve the page at the given the
+        # URL paths
+        default_site = Site.objects.get(is_default_site=True)
+        root_page = default_site.root_page
+
+        for path in url_paths:
+            path_components = path.split("/")
+
+            # Get the filterable list page we"re interested in
+            try:
+                filterable_page = root_page.route(None, path_components).page
+            except Http404:
+                raise CommandError(
+                    f"Unable to find a filterable list page for {path}. "
```

Comment:
This suggestion makes the error message more accurate, but the better fix is probably to check the page type if it doesn't 404. If the page is found but is not a filterable page, it throws "AttributeError: 'LandingPage' object has no attribute 'get_filterable_queryset'".

```suggestion
                    f"Unable to find a page at {path}. "
```

Reply:
Done in 1e70b31.